lib/shared/snackbar.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+
+enum SnackBarMode { normal, warning }
+
+void showSnackbar(BuildContext context, String text, {SnackBarMode mode = SnackBarMode.normal, bool clearSnackBars = true, Duration? duration, void Function()? undoAction}) {

```

Comment:
A couple of things!

1. `undoAction` could just be renamed to `action`, and we could allow for a custom icon for the `action` to make it more generic. Or an alternative could be to rename it to `trailing` and make it accept a `Widget` similar to how `AppBar` does it
3. I think I prefer `type` instead of `mode`. Then for the enum, it could be 

```dart
enum SnackBarType { info, warning }
```

Reply:
Good points. This could be made super generic by accepting a leading and a trailing widget. But I don't want the caller to do _too_ much work. 😆 So maybe a good compromise would be a leading icon, a trailing icon, and an action. That would also remove the need for the enum.